### ISSUE TITLE
[Audit] Validate modal a11y with Audit results

### ISSUE THREAD

**Issue by RCopeland** [sparkdesignsystem/Backlog#264]
**User Story:**
As Spark we want to confirm that our Modals follow the recommendations in the RM audit.

**Notes:** 
- This issues comes out of the RM Audit, RM-10. The examples that are called out as having errors don't appear to be using Spark. 

**AC:**
- read through the RM a11y audit (rm-10) and verify that spark's modal is compliant. 
- If not compliant, open issues with PO to make it so.

______________________________________________________________________________________

### Recommendations

The container for the dialog must be correctly identified using an appropriate role. In general, all dialogs can have a generic role of dialog.

The container for the dialog should also have the following attributes:

- aria-labelledby - used to identify the element containing the accessible name for the dialog (generally, the visible title for the dialog). If no appropriate visible title is present, aria-label="..." can be used to provide a non-visible title instead.
- aria-describedby (optional) - used to identify the element containing the main information/content of the dialog.

With all the relevant role and aria- attributes present, assistive technologies will announce the presence of the dialog (and, if available, its title and description) when the user's focus moves to/into the dialog.

## Recommended markup for the "Terms of Use" modal dialog (source: 01 - Create Account)

<div role=dialog" aria-labelledby="dialog-title" class="modal">
    <div class="modal-content">
        <h1 id="dialog-title">Rocket Account Terms of Use</h1>
        <span role=button" tabindex="0" aria-label="Close" class="closeBtn"></span>
               
    </div>
</div>

## Recommended markup for the "Chat With Us Now" modal dialog (source: 11 - Contact Us)

Note: the existing images that currently comprise this component have been replaced with structured content to provide a more accessible experience. Additional images, such as the Rocket Mortgage logo and the chat representative avatar may be added as required.

<div role=dialog" aria-labelledby="dialog-title" id="LPMcontainer-1571227041189-1" class="LPMcontainer LPMoverlay">
    <div id="LPMcontainer-1571227041201-2" role="button" tabindex="0">
        <img src=".../RM_Banker_Chat_Invite_Base.png" id="LPMimage-1571227041201-3" alt="Click to Live Chat">
    </div>
    <img src=".../RMA-chat-pop-close.png" id="LPMcloseButton-1571227041201-4" alt="Close" class="LPMcloseButton" role="button" tabindex="0">
    <h1 id="dialog-title">Chat With Us</h1>
    <p>Hi there! We need a little more info from you before we can move forward.</p>
    <button ...>Chat With Us Now</button>
</div>

When a dialog is launched, focus must be moved to the dialog. The specific element that should receive focus depends in large part on the purpose and structure of the dialog.

If in doubt, the safest option in all situations is to set focus to the actual dialog element itself. To achieve this, make sure that the dialog element can be programmatically focused (with the JavaScript focus() method) by setting tabindex="-1" on the element.

Example code: Using tabindex="-1" to ensure that the dialog can receive focus programmatically.

<div role="dialog" tabindex="-1" aria-label="dialog title goes here">
  ...
</div>

Both modal and non-modal dialogs should generally “contain” focus. Tab and Shift + Tab should cycle through the focusable controls in the dialog until the user either closes the dialog (by activating a confirmation button, using the "Close" button, or pressing the Esc key.

When a dialog is closed, focus should return to the element in the main document which originally had focus. In general, this would be the control or button that triggered the modal, unless the dialog was not shown as a direct result of a user action (for instance, a dialog that is shown as a timeout warning), in which case it would be the last element in the main part of the document that had focus. This latter point is also important when it comes to non-modal dialogs: when switching from the dialog back to the page, the element that last had focus must be re-focused. This means that whenever the user switches between the main document and the non-modal dialog (using F6), the site should keep track of the element that last had focus, and return focus to it once the user switches back.

When a modal dialog is displayed, any content that is not part of the modal dialog content must not be included in the focus order (since the dialog should "contain" focus). However, content should also be hidden from assistive technologies, to avoid that AT users navigate back into the underlying main document using alternative methods (such as the virtual cursor, navigation by headings, etc).

Example code: setting aria-hidden="true" on the container element for any non dialog content when the dialog is open

<body>
  <div aria-hidden="true">
    
  </div>
  <div role="dialog" class="visible">
    
  </div>
</body>

For a solid implementation of an accessible custom dialog, see A11y Dialog.

**Comment by ssabo1** [comment]
Put off till Spark Audit has been completed.
